app/pages/sections/order-items/catalogue-solutions/order-item/flat/ondemand/ui-tests/withSavedData.ui.test.js

```diff
@@ -0,0 +1,283 @@
+  orderItemId: 1,
+  serviceRecipient: recipients[0],
+  catalogueItemId: orderItem.catalogueItemId,
+  catalogueItemName: orderItem.catalogueItemName,
+  catalogueItemType: orderItem.catalogueItemType,
+  deliveryDate: orderItem.deliveryDate,
+  estimationPeriod: orderItem.estimationPeriod,
+};
+
+const mocks = () => {
+  nock(orderApiUrl)
+    .get('/api/v1/orders/order-id/order-items')
+    .reply(200, [orderItem]);
+};
+
+const defaultPageSetup = { withAuth: true, getRoute: true, postRoute: false };
+const pageSetup = async (setup = defaultPageSetup) => {
+  if (setup.withAuth) {
+    await setState(ClientFunction)('fakeToken', authTokenInSession);
+  }
+  if (setup.getRoute) {
+    mocks();
+  }
+  if (setup.postRoute) {
+    await setState(ClientFunction)(sessionKeys.orderItemPageData, orderItemPageDataInSession);
+  }
+};
+
+fixture('Catalogue-solutions - flat ondeman - withSavedData')
+  .page('http://localhost:1234/order/some-fake-page')
+  .afterEach(async (t) => {
+    await nockAndErrorCheck(nock, t);
+  });
+
+test('should render a text field for the price question', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const price = Selector('[data-test-id="question-price"]');
+  const priceSpan = price.find('span');
+  const priceLabel = price.find('label.nhsuk-label');
+
+  await t
+    .expect(await extractInnerText(priceLabel)).eql(content.questions.price.mainAdvice)
+    .expect(price.find('input').count).eql(1)
+    .expect(await extractInnerText(priceSpan)).eql('per consultation per annum');
+});
+
+test('should render the title', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const title = Selector('h1[data-test-id="order-item-page-title"]');
+
+  await t
+    .expect(await extractInnerText(title)).eql('Some item name information for order-id');
+});
+
+test('should link to /order/organisation/order-id/catalogue-solutions for backlink', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const goBackLink = Selector('[data-test-id="go-back-link"] a');
+
+  await t
+    .expect(goBackLink.getAttribute('href')).eql('/order/organisation/order-id/catalogue-solutions');
+});
+
+test('should populate text field for the price question', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const price = Selector('[data-test-id="question-price"] input');
+
+  await t
+    .expect(price.getAttribute('value')).eql('0.10');
+});
+
+test('should render the solution table content', async (t) => {
+  await pageSetup();
+  await t.navigateTo(pageUrl);
+
+  const table = Selector('div[data-test-id="solution-table"]');
+  const row = table.find('[data-test-id="table-row-0"]');
+  const solutionName = row.find('div[data-test-id="Some service recipient 2-OX3-recipient"]');
+  const quantityInput = row.find('[data-test-id="question-quantity"] input');
+  const quantityExpandableSection = row.find('[data-test-id="view-section-input-id-practice"]');
```

Comment:
We probably ought to rename this data test ID too as `practice` only applies to per patient prices, although it is probably worth checking the impact first – we might want to treat this as tech debt.